core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -505,7 +525,29 @@ public void refresh(String source) throws EngineException {
             throw new RefreshFailedEngineException(shardId, t);
         }
 
-        // TODO: maybe we should just put a scheduled job in threadPool?
+        mergeScheduler.refreshConfig();
```

Comment:
This is pre-existing, and I don't know why refresh means it's a good time to ask merge scheduler to pull any settings changes.  We already call this in `onSettingsChanged`... I suspect it's not really necessary :)